test/core/fields/operators_test.py

```diff
@@ -24,6 +25,53 @@ def test_args_fixture():
     return Args(C=C, parameter1=parameter1, parameter2=parameter2)
 
 
+def test_basic_gradient(test_args):
+    field = ScalarField(
+        lambda point: point.x**2 + point.y**2 - point.z**2
+    )
+    result_field = gradient_operator(field)
+    result_vector = result_field.apply_to_basis()
+    x = field.coordinate_system.coord_system.base_scalars()[0]
+    y = field.coordinate_system.coord_system.base_scalars()[1]
+    z = field.coordinate_system.coord_system.base_scalars()[2]
+    for i, expr in enumerate([2 * x, 2 * y, - 2 * z]):
+        assert expr_equals(result_vector.components[i], expr)
+
+
+def test_cylindrical_gradient(test_args):
+    # verify that in cylindrical coordinates the result is same
+    C1 = CoordinateSystem(CoordinateSystem.System.CYLINDRICAL)
+
+    def field_function(p: CylinderPoint) -> ScalarValue:
+        return p.r ** 2 - p.z ** 2
+
+    cylindrical_field = ScalarField(field_function, C1)
+    result_cylindrical_field = gradient_operator(cylindrical_field)
+    result_cylindrical_vector = result_cylindrical_field.apply_to_basis()
+
+    r = cylindrical_field.coordinate_system.coord_system.base_scalars()[0]
+    z = cylindrical_field.coordinate_system.coord_system.base_scalars()[2]
+    for i, expr in enumerate([2 * r, 0, - 2 * z]):
+        assert expr_equals(result_cylindrical_vector.components[i], expr)
+
+
+def test_spherical_gradient(test_args):
+    # verify that in spherical coordinates the result is the same
+    C1 = CoordinateSystem(CoordinateSystem.System.SPHERICAL)
+
+    def field_function(p: SpherePoint) -> ScalarValue:
+        return - p.r**2 * cos(2 * p.theta)
+
+    spherical_field = ScalarField(field_function, C1)
+    result_spherical_field = gradient_operator(spherical_field)
+    result_spherical_vector = result_spherical_field.apply_to_basis()
+
+    r = spherical_field.coordinate_system.coord_system.base_scalars()[0]
+    theta = spherical_field.coordinate_system.coord_system.base_scalars()[1]
+    for i, expr in enumerate([-2 * r * cos(2 * theta), 2 * r * sin(2 * theta), 0]):
```

Comment:
Where did you get the test value? How come it works despite the gradient operator is wrong?